### DISCUSSION TITLE
How to extract table of contents in PDF

### DISCUSSION THREAD

**Discussion by Ynjxsjmh** [jsvine/pdfplumber#481] [Q&A] [answered]
PyPDF2 has [getOutlines()](https://pythonhosted.org/PyPDF2/PdfFileReader.html#PyPDF2.PdfFileReader.getOutlines) function to extract the table of contents in PDF. Is there any similar function in pdfplumber? I read the README and search in issues and google but didn't find anything related.

**Comment by samkit-jain** [comment, marked as answer]
Hi @Ynjxsjmh `pdfplumber` is built on `pdfminer.six` and it also provides a `get_outlines(...)` method. It might be different from the one provided by `PyPDF2`. To access it, you can use the following code
```python
>>> import pdfplumber
>>> pdf = pdfplumber.open("file.pdf")
>>> pdf.doc.get_outlines()
```

`pdf.doc` is an instance of [PDFDocument](https://github.com/pdfminer/pdfminer.six/blob/33b60dfd5414bf91fa232550528973b07028ff64/pdfminer/pdfdocument.py#L508). An example on how to use the method can be found [here](https://github.com/pdfminer/pdfminer.six/blob/33b60dfd5414bf91fa232550528973b07028ff64/tools/dumppdf.py#L140).
